src/Sprites/ArtWriter.cpp

```diff
@@ -28,7 +29,16 @@ void ArtFile::WritePalettes(Stream::SeekableWriter& seekableWriter, const ArtFil
 	}
 
 	seekableWriter.Write(SectionHeader(TagPalette, static_cast<uint32_t>(artFile.palettes.size())));
-	seekableWriter.Write(PaletteHeader(artFile));
+
+	// Intentially do not pass palette as reference to allow local modification
+	// Switch red and blue color to match Outpost 2 custom format.
+	for (auto pallete : artFile.palettes) {
```

Comment:
I like this comment. This is a very good point about modifying a copy versus modifying in place. Though I'm a little uncertain what the comment is referring to here. Is this the use of `auto` without `&`? I suppose the array is of actual palette objects, and not simple references, so yes, I think you're correct here. Hmm, that's actually a nice way of handling it.

I think this warrants a test, where a write is performed and then the palette is checked to make sure it has not been modified. Alternatively, two back to back writes can be performed and checked they produce the same results.

Reply:
Thanks for looking this over. I think it is a little unusual way to solve the problem, hence the comment. 

I will try to add a test for this shortly as it should be pretty easy.

-Brett